### PR TITLE
issue 636 -Fixing Continuous Policy update in case of EXTENDED_Policy enaled

### PR DESCRIPTION
**Pull request by madhuy**
This fix is, suggestion made in PR #644 .

### CODE REVIEW

**Comment by jacobkeeler** [comment]
@madhuy Should #644 be closed in favor of this PR?

**Comment by madhuy** [comment]
@jacobkeeler Yes. You can close #644 . Could you please review it.

File: src/components/application_manager/src/application_manager_impl.cc [hunk 1]
```diff
@@ -1404,16 +1404,16 @@ void ApplicationManagerImpl::SendMessageToMobile(
...
+   /* if (function_id == mobile_apis::FunctionID::OnSystemRequestID) {
       mobile_apis::RequestType::eType request_type =
           static_cast<mobile_apis::RequestType::eType>(
               (*message)[strings::msg_params][strings::request_type].asUInt());
       if (mobile_apis::RequestType::PROPRIETARY == request_type ||
           mobile_apis::RequestType::HTTP == request_type) {
         GetPolicyHandler().OnUpdateRequestSentToMobile();
       }
-    }
+    }*/
```
Comment (jacobkeeler) [inline src/components/application_manager/src/application_manager_impl.cc]: @madhuy One small issue with this, the Travis build uses a style checking script (located [here](https://github.com/smartdevicelink/sdl_core/blob/master/tools/infrastructure/check_style.sh)) and is currently failing on this block, saying that the block should be moved forward by one space. 

```
$ bash -e tools/infrastructure/check_style.sh

<    /* if (function_id == mobile_apis::FunctionID::OnSystemRequestID) {
<       mobile_apis::RequestType::eType request_type =
<           static_cast<mobile_apis::RequestType::eType>(
<               (*message)[strings::msg_params][strings::request_type].asUInt());
<       if (mobile_apis::RequestType::PROPRIETARY == request_type ||
<           mobile_apis::RequestType::HTTP == request_type) {
<         GetPolicyHandler().OnUpdateRequestSentToMobile();
<       }
<     }*/
---
>     /* if (function_id == mobile_apis::FunctionID::OnSystemRequestID) {
>        mobile_apis::RequestType::eType request_type =
>            static_cast<mobile_apis::RequestType::eType>(
>                (*message)[strings::msg_params][strings::request_type].asUInt());
>        if (mobile_apis::RequestType::PROPRIETARY == request_type ||
>            mobile_apis::RequestType::HTTP == request_type) {
>          GetPolicyHandler().OnUpdateRequestSentToMobile();
>        }
>      }*/

The command "bash -e tools/infrastructure/check_style.sh" exited with 1.
```

You can run this script on your local machine from your sdl_core folder by running this command:

```bash
bash -e tools/infrastructure/check_style.sh
```

You may need to install `clang-format-3.6` beforehand using `sudo apt-get install clang-format-3.6`

**Comment by madhuy** [comment]
@jacobkeeler Sure I will run and update.

**Comment by madhuy** [comment]
@jacobkeeler  I am planning to get rid of dead code.

**Comment by madhuy** [comment]
@jacobkeeler Updated PR with changes.

**Comment by jacobkeeler** [comment]
Though we probably won't close this PR because of it, but for future reference please be sure to use the naming scheme of hotfix/branch_name when making PRs to the SDL repos.

**Comment by jacobkeeler** [comment]
@madhuy Any reason why the file permissions were changed for both the Mobile and HMI on_button_press_notification.cc files?

File: src/components/application_manager/src/application_manager_impl.cc [hunk 2]
```diff
@@ -1405,15 +1405,6 @@ void ApplicationManagerImpl::SendMessageToMobile(
       return;
     }
```
Comment (jacobkeeler) [inline src/components/application_manager/src/application_manager_impl.cc]: The style checker also has an issue with the extra newline which separates the two end braces, I should be able to fix this quickly through GitHub

Comment (madhuy) [inline src/components/application_manager/src/application_manager_impl.cc]: @jacobkeeler Thanks for fixing.

**Comment by madhuy** [comment]
@jacobkeeler Not sure reason for permission change for the files on_button_press_notification.cc. I will fallow the naming scheme .

**Comment by jacobkeeler** [comment]
@madhuy It looks like the file mode changes were made during some of your rebasing commits (https://github.com/smartdevicelink/sdl_core/pull/865/commits/7363c0b6d5d2d7b55a9975a9e53cc0a27a9be01e contains only a change of this type), please revert these changes.

**Comment by jacobkeeler** [comment]
@mghiumiusliu Please assign someone to review

**Comment by madhuy** [comment]
@mghiumiusliu Could you please review ?

**Comment by madhuy** [comment]
@mghiumiusliu Could you please prioritize this PR for review ?

**Comment by JackLivio** [comment]
@LuxoftAKutsan Could you maybe take a look at this PR, please?

**Comment by jacobkeeler** [comment]
@madhuy Please remove the mode change for [mobile/on_button_press_notification.cc](https://github.com/madhuy/sdl_core/blob/2617a208cf5bd619d39996cb7d08dd3756dc970f/src/components/application_manager/src/commands/mobile/on_button_press_notification.cc) as well

**Pull request merged**